Scripts/Restriction/ranacompiler.py

```diff
@@ -486,54 +487,40 @@ def build_dict(self):
         with open(os.path.join(update, "Restriction_Dictionary.py"), "w") as results:
             print("Writing the dictionary containing the new Restriction classes...")
             results.write(start)
-            results.write("# Turn black code style off\n# fmt: off\n")
             results.write("rest_dict = {}\n")
-            results.write("\n\n")
+            results.write("\n")
             for name in sorted(classdict):
-                results.write("def _temp():\n")
-                results.write("    return {\n")
+                results.write("rest_dict[%s] = {\n" % double_quote_repr(name))
                 for key, value in sorted(classdict[name].items()):
                     results.write(
-                        "        %s: %s,\n"
+                        "    %s: %s,\n"
                         % (double_quote_repr(key), double_quote_repr(value))
                     )
-                results.write("    }\n")
-                results.write("\n\n")
-                results.write("rest_dict[%s] = _temp()\n" % double_quote_repr(name))
-                results.write("\n\n")
+                results.write("}\n\n")
             print("OK.\n")
             print("Writing the dictionary containing the suppliers data...")
+            results.write("\n")
+            results.write("# Turn black code style off\n# fmt: off\n")
+            results.write("\n")
             results.write("suppliers = {}\n")
-            results.write("\n\n")
+            results.write("\n")
             for name in sorted(suppliersdict):
-                results.write("def _temp():\n")
-                results.write("    return (\n")
+                results.write("suppliers[%s] = (\n" % double_quote_repr(name))
                 for value in suppliersdict[name]:
-                    results.write("        %s,\n" % double_quote_repr(value))
-                results.write("    )\n")
-                results.write("\n\n")
-                results.write("suppliers[%s] = _temp()\n" % double_quote_repr(name))
-                results.write("\n\n")
+                    results.write("    %s,\n" % double_quote_repr(value))
+                results.write(")\n\n")
             print("OK.\n")
             print("Writing the dictionary containing the Restriction types...")
+            results.write("\n")
             results.write("typedict = {}\n")
-            results.write("\n\n")
+            results.write("\n")
             for name in sorted(typedict):
-                results.write("def _temp():\n")
-                results.write("    return (\n")
+                results.write("typedict[%s] = (\n" % double_quote_repr(name))
                 for value in typedict[name]:
```

Comment:
As above, according to `Restriction_Dictionary.py`, `typedict[name]` has only one value (a list)? So no need for the `for` loop here?